src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,151 @@
-
+        if (table == null) {
+            resize();
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        if (hash < 0) {
+            hash *= -1;
+        }
+        int position = hash % table.length;
+        if (table[position] == null) {
+            table[position] = new Node<>(hash, key, value, null);
+        } else {
+            // check of existing nodes
+            Node<K, V> current = table[position];
+            if (current.key == key || current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            while (current.next != null) {
+                if (current.key == key) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            // check the last node
+            if (current.key == key || current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            }
+            current.next = new Node<>(hash, key, value, null);
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (table == null) {
+            return null;
+        }
+        int hash = key == null ? 0 : key.hashCode();
+        if (hash < 0) {
+            hash *= -1;
+        }
+        int position = hash % table.length;
+        Node<K, V> current = table[position];
+        while (!(current.key == key || current.key != null && current.key.equals(key))) {
+            current = current.next;
+            if (current == null) {
+                break;
+            }
+        }
+        if (current == null) {
+            return null;
+        }
+        return current.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTab = table;
+        int oldCap = (oldTab == null) ? 0 : oldTab.length;
+        int newCap;
+        int newThr;
+        Node<K, V>[] newTab;
+        if (oldCap == 0) {
+            table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+            threshold = (int) (DEFAULT_LOAD_FACTOR * table.length);
+        } else {
+            newCap = oldTab.length * 2;
+            newThr = (int) (DEFAULT_LOAD_FACTOR * newCap);
+            newTab = (Node<K, V>[]) new Node[newCap];
+            threshold = newThr;
+            for (Node<K, V> node : table) {
+                if (node != null) {
+                    while (node.next != null) {
+                        putNode(newTab, node, newCap);
+                        node = node.next;
+                    }
+                    putNode(newTab, node, newCap);
+                }
+            }
+            table = newTab;
+        }
+    }
+
+    private void putNode(Node<K, V>[] table, Node<K,V> node, int capacity) {
```

Comment:
The method 'putNode' is only used inside the 'MyHashMap' class and should have a 'private' access modifier to adhere to encapsulation principles.